test/one/profiler/test/Runner.java

```diff
@@ -0,0 +1,242 @@
+                    return Jvm.J9;
+                }
+            }
+        }
+
+        // Zing check
+
+        // Workaround for java.home including /jre in JDK 8
+        if (currentJvmVersion == 8) {
+            javaHome = javaHome.getParentFile();
+        }
+
+        // Workaround for contents/home on mac
+        if (currentOs == Os.MACOS) {
+            javaHome = javaHome.getParentFile();
+        }
+        File etcDirectory = new File(javaHome, "etc");
+        File zing = new File(etcDirectory, "zing");
+        if (zing.exists()) {
+            return Jvm.ZING;
+        }
+
+        // Otherwise it's some variation of hotspot
+        return Jvm.HOTSPOT;
+    }
+
+    private static Arch detectArch() throws RuntimeException{
+        String architecture = System.getProperty("os.arch");
+
+        if (architecture.contains("x86_64") || architecture.contains("amd64")) {
+            return Arch.X64;
+        } else if (architecture.contains("aarch64")) {
+            return Arch.ARM64;
+        } else if (architecture.contains("arm")) {
+            return Arch.ARM32;
+        } else if (architecture.contains("ppc64le")) {
+            return Arch.PPC64LE;
+        } else if (architecture.endsWith("86")) {
+            return Arch.X86;
+        } else {
+            throw new RuntimeException("Unable to detect the OS used by the host.\nOS may be unsupported.");
+        }
+    }
+
+    public static boolean enabled(Test test) {
+        Os[] os = test.os();
+        int[] jdkVersion = test.jdkVersion();
+        Jvm[] jvm = test.jvm();
+        Arch[] arch = test.arch();
+
+        if (!test.enabled()) {
+            return false;
+        }
+
+        if (os.length > 0 && !Arrays.asList(os).contains(currentOs)) {
+            return false;
+        }
+
+        if (jdkVersion.length > 0 && !Arrays.asList(jdkVersion).contains(currentJvmVersion)) {
+            return false;
+        }
+
+        if (jvm.length > 0 && !Arrays.asList(jvm).contains(currentJvm)) {
+            return false;
+        }
+
+        if (arch.length > 0 && !Arrays.asList(arch).contains(currentArch)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public static boolean run(Method m, Set skipSet, boolean retainlogs) throws Exception{
+        boolean passedAll = true;
+        for (Test test : m.getAnnotationsByType(Test.class)) {
+            String fullDeclaringClass = m.getDeclaringClass().getName();
+            String className = fullDeclaringClass.substring(fullDeclaringClass.lastIndexOf('.') + 1);
+
+            if (!enabled(test) || skipSet.contains(m.getName().toLowerCase()) || skipSet.contains(className.toLowerCase())) {
+                log.log(Level.FINE, "Skipped " + m.getDeclaringClass().getName() + '.' + m.getName());
+                continue;
+            }
+
+            String testName = null;
+            if (retainlogs) {
+                testName = className + '.' + m.getName();
```

Comment:
Why should `testName` depend on `retainlogs`?

Reply:
Removed the condition.